node/caching-compiler.js

```diff
@@ -17,27 +17,32 @@ const reifyVersion = require("../package.json")
 
 const DEFAULT_CACHE_DIR = ".reify-cache";
 
-exports.compile = (content, filename) => {
-  if (filename === "repl") {
+exports.compile = (content, options) => {
+  options = options || Object.create(null);
+
+  if (options.filename === "repl") {
     // Treat the REPL as if there was no filename.
-    filename = null;
+    options.filename = null;
   }
 
-  const pkgInfo = filename ? getPkgInfo(filename) : fallbackPkgInfo;
+  const pkgInfo = options.filename
+    ? getPkgInfo(options.filename)
+    : fallbackPkgInfo;
+
   if (! pkgInfo) {
     return content;
   }
 
-  return filename
-    ? readWithCacheAndFilename(pkgInfo, content, filename)
-    : readWithCache(pkgInfo, content);
+  return options.filename
+    ? readWithCacheAndFilename(pkgInfo, content, options)
+    : readWithCache(pkgInfo, content, options);
 };
 
-function readWithCacheAndFilename(pkgInfo, content, filename) {
+function readWithCacheAndFilename(pkgInfo, content, options) {
   try {
```

Comment:
When I was working with this for the PR I didn't really dig the names of the methods. I was like `readWithCacheAnd....` then I saw it was also writing the file and compiling. I dig the separation to allow try-catch for just the case where a filename is known but didn't bikeshed a better name yet.